setup.py

```diff
@@ -266,14 +265,142 @@ def extensions():
 
 
 class install(orig_install.install):
+    description = "Installs dpctl into Python prefix"
+    user_options = orig_install.install.user_options + [
+        (
+            "level-zero-support=",
+            None,
+            "Whether to enable support for program creation "
+            "for Level-zero backend",
+        ),
+        (
+            "sycl-compiler-prefix=",
+            None,
+            "Path to SYCL compiler installation. None means "
+            "read it off ONEAPI_ROOT environment variable or fail.",
+        ),
+        (
+            "coverage=",
+            None,
+            "Whether to generate coverage report "
+            "when building the backend library",
+        ),
+    ]
+
+    def initialize_options(self):
+        super().initialize_options()
+        self.level_zero_support = "True"
+        self.coverage = os.getenv("CODE_COVERAGE", "False")
```

Comment:
let us just remove the ennvar to control generation of coverage data?

Reply:
Environment variable  ``CODE_COVERAGE`` is also used to pass additional Cython extensions defines.

I think option ``--coverage`` only makes sense for `develop` command, not `install`.

Reply:
I agree. The `--coverage` option forces a `DEBUG` build for the C API and we should not allow the option with `install`. How and when does `setuptools` call the `extension` function? Is it not possible to check the `--coverage` option that was provided to `setup.py`in `extension`?

Reply:
The `extensions()` is called during call to `setup`, way before arguments are processed. It is possible to pass the define via `build_ext` command see https://medium.com/@dfdeshom/better-test-coverage-workflow-for-cython-modules-631615eb197a

I have to try if that works.